File: spring-java-starter/spring-java-mysql/starter/src/test/java/io/github/mrgsrylm/builder/BaseBuilder.java
package io.github.mrgsrylm.builder;

import lombok.SneakyThrows;

public abstract class BaseBuilder<T> {
    T data;

    @SneakyThrows
    public BaseBuilder(Class<T> clazz) {
        this.data = clazz.getDeclaredConstructor().newInstance();
    }

    public T build() {
        return data;
    }
}

File: spring-kotlin-starter/spring-kotlin-mysql/starter/src/test/kotlin/io/github/mrgsrylm/builder/BaseBuilder.kt
package io.github.mrgsrylm.builder

abstract class BaseBuilder<T> {
    abstract var data: T;

    fun BaseBuilder(clazz: Class<T>) {
        this.data = clazz.getDeclaredConstructor().newInstance()
    }
    
    fun build(): T {
        return data
    }
}